activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -506,7 +525,13 @@ def find_some(ids)
       def find_some_ordered(ids)
         ids = ids.slice(offset_value || 0, limit_value || ids.size) || []
 
-        relation = except(:limit, :offset).where(primary_key => ids)
+        relation = except(:limit, :offset)
+        relation = if klass.composite_primary_key?
+          or_scope = ids.map { |values_set| where(primary_key.zip(values_set).to_h) }.inject(&:or)
+          relation.merge(or_scope)
```

Comment:
merge is scary... is `ids.map { relation.where(..) }.inject(:or)` not a good fit here?

Reply:
I clearly missed something in my reasoning as `user_type='admin'` doesn't get duplicated if we just use `relation.where(..)` 

So I simplified it as per your suggestion

Reply:
I will revisit 
https://github.com/rails/rails/blob/4bd3a108cc245383e8b59ea6ae7937ecb3853587/activerecord/lib/active_record/associations/preloader/association.rb#L40-L41

At least to consider using `scope.and` instead of `scope.merge` but I do remember having some issues with using `scope.where` instead of `scope.klass.where` (clean state)

Eventually I'm hoping we can define an internal helper to build these cross-tenant `OR` statements and reuse it everywhere. But I've been postponing this until we have more use-cases